tests/unit-core/test-api-set-and-clear-error-flag.c

```diff
@@ -31,6 +31,51 @@ main (void)
 
   JERRY_ASSERT (obj_val != err_val);
   jerry_release_value (err_val);
+  jerry_release_value (obj_val);
+
+  jerry_value_t str = jerry_create_string ((jerry_char_t *) "Pterodactylus");
+  jerry_size_t str_size = jerry_get_utf8_string_size (str);
+  jerry_char_t str_buff[str_size];
+  jerry_string_to_utf8_char_buffer (str, str_buff, str_size);
+  str_buff[str_size] = '\0';
+  jerry_value_t error = jerry_create_error_from_value (str, true);
+  str = jerry_get_value_from_error (error, true);
+  str_size = jerry_get_string_size (str);
+  jerry_char_t str_buff2[str_size];
+  jerry_string_to_utf8_char_buffer (str, str_buff2, str_size);
+  str_buff2[str_size] = '\0';
+  JERRY_ASSERT (!strcmp ((char *) str_buff, (char *) str_buff2));
+  jerry_release_value (str);
+
+  str = jerry_create_string ((jerry_char_t *) "Pterodactylus");
+  str_size = jerry_get_utf8_string_size (str);
+  jerry_string_to_utf8_char_buffer (str, str_buff, str_size);
+  str_buff[str_size] = '\0';
+  error = jerry_create_error_from_value (str, false);
+  jerry_release_value (str);
+  str = jerry_get_value_from_error (error, true);
+  str_size = jerry_get_string_size (str);
+  jerry_string_to_utf8_char_buffer (str, str_buff2, str_size);
+  str_buff2[str_size] = '\0';
+  JERRY_ASSERT (!strcmp ((char *) str_buff, (char *) str_buff2));
+  jerry_release_value (str);
+
+  jerry_value_t num = jerry_create_number (3.1415926);
+  double dNum = jerry_get_number_value (num);
+  jerry_value_t num2 = jerry_create_error_from_value (num, true);
+  num2 = jerry_get_value_from_error (num2, true);
+  double dNum2 = jerry_get_number_value (num2);
+  JERRY_ASSERT (dNum == dNum2);
+  jerry_release_value (num2);
+
+  num = jerry_create_number (3.1415926);
+  dNum = jerry_get_number_value (num);
```

Comment:
Perhaps you can simply use a local variable here:

```
double test_num = 3.1415926
jerry_value_t num = jerry_create_number (test_num);
jerry_value_t num2 = jerry_create_error_from_value (num, false);
JERRY_ASSERT (jerry_value_is_error (num2));
jerry_release_value (num);
num2 = jerry_get_value_from_error (num2, true);
JERRY_ASSERT (jerry_get_number_value (num2) == test_num);
jerry_release_value (num2);
```

The other test cases should be updated as well.

Reply:
This could be an interesting test case as well:

```
double test_num = 3.1415926
jerry_value_t num = jerry_create_number (test_num);
jerry_value_t num2 = jerry_create_error_from_value (num, false);
JERRY_ASSERT (jerry_value_is_error (num2));
jerry_release_value (num);
jerry_value_t num3 = jerry_create_error_from_value (num2, false);
JERRY_ASSERT (jerry_value_is_error (num3));
jerry_release_value (num2);
num2 = jerry_get_value_from_error (num3, true);
JERRY_ASSERT (jerry_get_number_value (num3) == test_num);
jerry_release_value (num3);
```

With both `jerry_create_error_from_value (num2, false);` and `jerry_create_error_from_value (num2, true);`